### ISSUE TITLE
Have groupings of pairs and different roi/stoploss/trailing per pair group

### ISSUE THREAD

**Issue by synackSA** [freqtrade/freqtrade#5123]
So I had a thought for feature. Right now it's hard to get a strategy that works will for all pairs since the volatility on pairs is all different, so an ROI and Stop loss that works well on one pair, could be really poor on another, even though the the basic strategy is valid.

So right now, this is how the params for a strategy might look:
```python
# Buy hyperspace params:
buy_params = {
    "buy_adx_dmp_threshhold": 30,
    "buy_adx_threshhold": 45,
}

# Sell hyperspace params:
sell_params = {
    "buy_adx_dmn_threshhold": 34,
    "buy_adx_threshhold": 32,
}

# ROI table:
minimal_roi = {
    "0": 0.38,
    "195": 0.162,
    "455": 0.076,
    "736": 0
}

# Stoploss:
stoploss = -0.164

# Trailing stop:
trailing_stop = True
trailing_stop_positive = 0.346
trailing_stop_positive_offset = 0.358
trailing_only_offset_is_reached = False
```

What I was thinking, is it could look something like this:
```python
pair_groups = {
    group_1: ["BTC/USDT", "ETH/USDT", "MTL/USDT"],
    group_2: ["BTH/USDT", "WRX/USDT", "DATA/USDT"],
    # etc
}
```

And then you could do something like either of these two ways
```python
buy_params: {
    group_1: {
        buy_adx_dmp_threshhold: 30,
        buy_adx_threshhold: 45,
    },
    group_2: {
      buy_adx_dmp_threshhold: 34,
      buy_adx_threshhold: 43,
    },
}
# Same for all the other params
```

Or 
```python
pair_params = {
    group_1: {
      buy_params: {
          "buy_adx_dmp_threshhold": 30,
          "buy_adx_threshhold": 45,
      },
      sell_params: {
          "buy_adx_dmn_threshhold": 34,
          "buy_adx_threshhold": 32,
      },
      minimal_roi: {
          "0": 0.38,
          "195": 0.162,
          "455": 0.076,
          "736": 0
      },
      trailing: {
        "stoploss": -0.164,
        "trailing_stop": true,
        "trailing_stop_positive": 0.346,
        "trailing_stop_positive_offset": 0.358,
        "trailing_only_offset_is_reached": false
      }
    },
    group_2: {
      # etc
    }
}
```

The idea would be to hyperopt as per normal, identify a group of pairs that do well under a set of params, save the pairs and params and remove the pairs from the pair list and then hyperopt again. Repeat this process until you hopefully have several groups of params that work well. This would remove the need to run multiple instances of a strategy with different pair lists and hopefully increase profitability.

I am aware that trying to hyperopt would only really work on the original format of the params and pair list, but I think that's fine personally and a trade off most would probably be willing to make.

I'm personally going to try and implement something like this into my strategy by overwriting some of the strategy functions to be able to handle this format, but I thought I would share the idea to see if it's something that could potentially be incorporated into Freqtrade.

**Comment by xmatthias** [comment]
You can do this pretty easily by running multiple bots.

Also within the strategy, you can simply use different parameternames per group and then filter per pair, and use custom stoploss to handle different stop values.


Obviously, youre free to implement and submit a PR trying to implement this, but i think it will complicate the code too much, which will reduce maintainability and therefore introduce subtle bugs, and therefor may end to be non mergable (speak, ill for sure look at it, but wont guarantee to incorporate / support this in the future).
